sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_client.py

```diff
@@ -98,7 +99,91 @@ def send_request(
     async def close(self) -> None:
         await self._client.close()
 
-    async def __aenter__(self) -> "EventGridClient":
+    async def __aenter__(self) -> "EventGridPublisherClient":
+        await self._client.__aenter__()
+        return self
+
+    async def __aexit__(self, *exc_details: Any) -> None:
+        await self._client.__aexit__(*exc_details)
+
+
+class EventGridConsumerClient(
+    EventGridConsumerClientOperationsMixin
+):  # pylint: disable=client-accepts-api-version-keyword
+    """EventGridConsumerClient.
+
+    :param endpoint: The host name of the namespace, e.g.
+     namespaceName1.westus-1.eventgrid.azure.net. Required.
+    :type endpoint: str
+    :param credential: Credential used to authenticate requests to the service. Is either a
+     AzureKeyCredential type or a TokenCredential type. Required.
+    :type credential: ~azure.core.credentials.AzureKeyCredential or
+     ~azure.core.credentials_async.AsyncTokenCredential
+    :keyword api_version: The API version to use for this operation. Default value is "2024-06-01".
+     Note that overriding this default value may result in unsupported behavior.
+    :paramtype api_version: str
+    """
+
+    def __init__(
+        self, endpoint: str, credential: Union[AzureKeyCredential, "AsyncTokenCredential"], **kwargs: Any
+    ) -> None:
+        _endpoint = "{endpoint}"
```

Comment:
does this value get replaced somewhere later?

Reply:
gen code -> _endpoint doesn't get replaced but this is so both _endpoint and endpoint can be passed in below I believe